feat(goose): add a migration registration helper allowing to use a pgx connection instead of an sql.DB instance (#908)
Scalingo/go-utils: postgresql/goose/README.md

# Goose

> Goose is a package that provide some useful functions to interact with a Postgres database.

Reviewer: Please always use "PostgreSQL" to refer to PostgreSQL. This is important that we stick with one way of naming this database type everywhere.

```suggestion
Goose is a package that provides some useful functions to interact with a PostgreSQL database.
```

### Add a migration using a pgx connection as parameter

> This package provide a function to register a migration that use a context and a pgx connection as parameter: `AddPgxContextMigration`

Reviewer: ```suggestion
This package provides a function to register a migration that use a context and a `pgx` connection as parameter: `AddPgxContextMigration`
```

> It allows the use of an sqlc querier to interact with the database easily:

Reviewer: suggestion: first mention of sqlc, I would add a link.
```suggestion
It allows the use of an [sqlc](https://sqlc.dev/) querier to interact with the database easily:
```

```go

import (
    ...
    "github.com/Scalingo/go-utils/postgresql/goose"
    ...
)

func init() {
	err := goose.AddPgxContextMigration(upCreateNode, downCreateNode)
	if err != nil {
		panic(err)
	}
}

func upMigration(ctx context.Context, conn *pgx.Conn) error {

>     querier := db.New(conn)

Reviewer: question: what is `db` here? I assume it's a package but it's unclear.
